sbt-dotty/sbt-test/source-dependencies/subproject-dependency-b/build.sbt

```diff
@@ -1,14 +1,28 @@
-lazy val commonSettings = Seq(
-  logLevel := Level.Debug
-)
-lazy val provider = project.settings(commonSettings)
-lazy val use = project.settings(commonSettings).dependsOn(provider)
+import sbt.internal.inc.Analysis
+import complete.DefaultParsers._
 
-InputKey[Unit]("check-number-of-compiler-iterations") <<= inputTask { (argTask: TaskKey[Seq[String]]) =>
-  (argTask, compile in Compile in use) map { (args: Seq[String], a: sbt.inc.Analysis) =>
-    assert(args.size == 1)
-    val expectedIterationsNumber = args(0).toInt
-    assert(a.compilations.allCompilations.size == expectedIterationsNumber, 
-      "a.compilations.allCompilations.size = %d (expected %d)".format(a.compilations.allCompilations.size, expectedIterationsNumber))
+// Reset compiler iterations, necessary because tests run in batch mode
+val recordPreviousIterations = taskKey[Unit]("Record previous iterations.")
```

Comment:
This task is duplicated throughout the codebase. It's fine to get it in to unblock other stuff dependent on it, but we should create an issue to DRY it out in the future.

Reply:
Maybe a compiler plugin would help here?

Reply:
The problem is to figure out how to inject a compiler plugin in the scripted tests, one way would be to have it as a separate project that we publish before we run the scripted tests, but that's pretty heavy-weight.